File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/quickstart/QuickstartSpec.java
package io.micronaut.rabbitmq.docs.quickstart;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;
import io.micronaut.rabbitmq.testcontainers.RabbitMQ;
import org.junit.jupiter.api.TestInstance;

@MicronautTest
@Property(name = "spec.name", value = "QuickstartSpec")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class QuickstartSpec implements TestPropertyProvider {
    @Override
    public Map<String, String> getProperties() {
        return RabbitMQ.getProperties();
    }

    @Test
    void testProductClientAndListener(ProductClient productClient, ProductListener productListener) {
// tag::producer[]
productClient.send("quickstart".getBytes());
// end::producer[]

        await().atMost(60, SECONDS).until(() ->
                productListener.messageLengths.size() == 1 &&
                productListener.messageLengths.get(0).equals("quickstart")
        );
    }
}

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/quickstart/QuickstartSpec.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.rabbitmq.docs.Await import Await
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .ProductClient import ProductClient
from .ProductListener import ProductListener


@MicronautTest(environments=["rabbitmq"])
@Property(name="spec.name", value="QuickstartSpec")
class QuickstartSpec:
    product_client: Annotated[ProductClient, Inject]
    product_listener: Annotated[ProductListener, Inject]

    @Test
    def test_product_client_and_listener(self) -> None:
# tag::producer[]
        self.product_client.send(b"quickstart")
# end::producer[]

        Await.until(lambda: self.product_listener.message_lengths == ["quickstart"])
